src/Abp.NHibernate/NHibernate/Uow/NhUnitOfWork.cs

```diff
@@ -117,28 +116,22 @@ public override Task SaveChangesAsync()
         protected override void CompleteUow()
         {
             SaveChanges();
-            if (_transaction != null)
-            {
-                _transaction.Commit();
-            }
+            _transaction?.Commit();
         }
 
-        protected override Task CompleteUowAsync()
+        protected override async Task CompleteUowAsync()
         {
-            CompleteUow();
-            return Task.FromResult(0);
+            await SaveChangesAsync();
+            await _transaction?.CommitAsync();
```

Comment:
Just reviewed this !!
Null propagation after await is definitely a bad idea. Sorry for this I will fix it.

Reply:
Fixed NullReferenceException and added unit tests